### ISSUE TITLE
Add SScanSS data parser

### ISSUE THREAD

**Issue by AndrewLister-STFC** [fitbenchmarking/fitbenchmarking#1040]
SScanSS is a slightly different problem to the standard FitBenchmarking problem as follows:

Given a robot <img src="https://render.githubusercontent.com/render/math?math=r_i">, parameters <img src="https://render.githubusercontent.com/render/math?math=p">, and a target <img src="https://render.githubusercontent.com/render/math?math=t">, solve <img src="https://render.githubusercontent.com/render/math?math=min_{p_i}\lvert r_i(p_i) - t\rvert^2">.

Functionally, the robot is considered to be a series of connected joints, either rotational of translational, with fixed axes and bounds on the range of motion, where the parameters represent the amount of rotation or translation. As joints earlier in the series move, this will affect all of the joints' positions further down the series. The solution is found when the position of the last joint of the robot reaches the desired position and orientation - the target position. Any position in the system is represented by a 3-vector for location and a quaternian for orientation. There may be multiple solutions but any will suffice.

One additional note is that the error being minimised is a slightly more complicated function where the location is differenced as expected, but the orientation is not a simple subtraction -- the rotation is first converted to 3 scalar rotations along the axes then the difference is taken.

The main difference here is that we have no input data to the robot, and that the output from f is multidimensional. For example, a usual fitbenchmarking problem (in the simplest case) would be:
<img src="https://render.githubusercontent.com/render/math?math=min_p\lvert f(x,p) - y \rvert^2">
Where x and y have the same number of elements.

If we treat the target as our y, then the norm will require some attention to ensure the difference in the orientation parts are handled correctly -- alternatively this could be bundled into the definition of f and y could be set to 0.

Another practical issue to overcome is the intended use and outputs. Where one problem for our existing applications requires a single data file with many corroborating data points to which we wish to aquire an estimate of some hidden property in noisy data. Inverse kinematics seeks to solve many positions for a single robot, by definition these will all have different solutions but would be of little use as individual results as the aim is to find a solver that is robust across the whole range of the robot -- perhaps it's best to consider a single problem as a full set of targets for a robot in the outputs from FitBenchmarking.

A demo of the inverse kinematics problem from SScanSS can be found here: https://bitbucket.org/SteveNneji/demo-for-fitbenchmarking/src/master/main.py

**Comment by AndrewLister-STFC** [comment]
@StephenNneji thanks for the clarification in the first 2 points.

Please can you remind me if joint offsets refers to the anchors and links or the parameters to find? -- for this I will consider it to be the latter.

For point 3, I had considered the joint offsets to be p which I'd refer to as the parameters. "Input data" was refering to the given x data where for an optimal set of parameters p we would have a minimum for:
sum_i [f(x_i, p) - y_i]^2

Here, it may make more sense to consider the x as the joint offsets as you say, p as the anchors and links, and y as the target poses, such that we still find the minimum as defined above but with p invariant and x as the unknown. Unfortunately all of our code is set up to optimise for p given x.

**Comment by StephenNneji** [comment]
@AndrewLister-STFC  Yes, the joint offsets are the parameters to find. Thanks for clarifying point 3
